server/vulnerabilities/msrc/analyzer_test.go

```diff
@@ -128,3 +130,83 @@ func TestAnalyzer(t *testing.T) {
 		})
 	})
 }
+
+func TestWinBuildVersionGreaterOrEqual(t *testing.T) {
+	tc := []struct {
+		name       string
+		feed       string
+		os         string
+		result     bool
+		errMessage string
+	}{
+		{
+			name:       "equal",
+			feed:       "10.0.22000.795",
+			os:         "10.0.22000.795",
+			result:     true,
+			errMessage: "",
+		},
+		{
+			name:       "greater",
+			feed:       "10.0.22000.795",
+			os:         "10.0.22000.796",
+			result:     true,
+			errMessage: "",
+		},
+		{
+			name:       "less",
+			feed:       "10.0.22000.795",
+			os:         "10.0.22000.794",
+			result:     false,
+			errMessage: "",
+		},
+		{
+			name:       "too many parts in feed version",
+			feed:       "10.0.22000.795.9999",
+			os:         "10.0.22000.794",
+			result:     false,
+			errMessage: "invalid feed version",
+		},
+		{
+			name:       "too many parts in os version",
+			feed:       "10.0.22000.795",
+			os:         "10.0.22000.794.9999",
+			result:     false,
+			errMessage: "invalid os version",
+		},
+		{
+			name:       "too few parts in feed version",
+			feed:       "10.0.22000",
+			os:         "10.0.22000.794",
+			result:     false,
+			errMessage: "invalid feed version",
+		},
+		{
+			name:       "empty feed version",
+			feed:       "",
+			os:         "10.0.22000.794",
+			result:     false,
+			errMessage: "empty feed version",
+		},
+		{
+			name:       "comparing different product versions",
+			feed:       "10.0.22000.795",
+			os:         "10.0.22621.795",
```

Comment:
What is `22000` and `22621`?

Reply:
22000 == Windows 11 21H2
22621 == Windows 11 22H2

If Product matching is occurring correctly we should never hit this scenario.  Addressing it here in case this function gets used elsewhere in the future.